python-package/xgboost/spark/core.py

```diff
@@ -1027,25 +1042,52 @@ def predict_udf(iterator: Iterator[pd.DataFrame]) -> Iterator[pd.Series]:
                 else:
                     base_margin = None
 
+                data = {}
                 preds = model.predict(
                     X,
                     base_margin=base_margin,
                     validate_features=False,
                     **predict_params,
                 )
-                yield pd.Series(preds)
+                data["prediction"] = pd.Series(preds)
```

Comment:
Can it handle multiple prediction types? For instance, normal prediction + contribs at the same time

Reply:
do you mean handling "normal prediction + contribs" in the same "predict" of XGBModel or in a single pandas udf?

Right now, this PR can handle the latter, can predict the "normal prediction + contribs" in a single pandas udf. If "predict" of XGBModel can support predicting multiple prediction types, I can change it accordingly.